osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/utils/ObjectConfigServiceHelper.java

```diff
@@ -34,32 +38,44 @@ private ObjectConfigServiceHelper(final ObjectConfigService objectConfigService)
    * Find an Attribute from the ObjectConfigService based on the protocol and protocolVersion and a
    * DlmsObjectType name. When not found the Optional.empty is returned.
    *
-   * @param protocol protocol like DSMR or SMR
-   * @param protocolVersion like 4.2.2 or 5.5
+   * @param protocol protocol like DSMR or SMR 5.5
    * @param dlmsObjectType the DlmsObjectType to find
-   * @param attributeId the attributeId to find
-   * @return Optional<AttributeAddress> when found it returns a newly created AttributeAddress of
+   * @return Optional<AttributeAddress> when found it returns a newly created AttributeAddress or
    *     else Optional.empty()
    */
   public Optional<AttributeAddress> findOptionalAttributeAddress(
-      final String protocol,
-      final String protocolVersion,
-      final DlmsObjectType dlmsObjectType,
-      final int attributeId)
+      final Protocol protocol, final DlmsObjectType dlmsObjectType)
+      throws ProtocolAdapterException {
+
+    return this.findOptionalAttributeAddress(protocol, dlmsObjectType, null);
+  }
+
+  /**
+   * Find an Attribute from the ObjectConfigService based on the protocol and protocolVersion and a
+   * DlmsObjectType name. When not found the Optional.empty is returned.
+   *
+   * @param protocol protocol like SMR 5.5
+   * @param dlmsObjectType the DlmsObjectType to find
+   * @param channel the channel of the device
+   * @return Optional<AttributeAddress> when found it returns a newly created AttributeAddress or
+   *     else Optional.empty()
+   */
+  public Optional<AttributeAddress> findOptionalAttributeAddress(
+      final Protocol protocol, final DlmsObjectType dlmsObjectType, final Integer channel)
       throws ProtocolAdapterException {
 
     final Optional<CosemObject> optObject =
-        this.getOptionalCosemObject(protocol, protocolVersion, dlmsObjectType);
+        this.getOptionalCosemObject(protocol.getName(), protocol.getVersion(), dlmsObjectType);
     if (optObject.isEmpty()) {
       return Optional.empty();
     }
 
     final CosemObject cosemObject = optObject.get();
     final int classId = cosemObject.getClassId();
-    final ObisCode obisCode = new ObisCode(cosemObject.getObis());
+    final ObisCode obisCode = this.replaceChannel(cosemObject.getObis(), channel);
 
     final Optional<Attribute> attributeOpt =
-        Optional.ofNullable(cosemObject.getAttribute(attributeId));
+        Optional.ofNullable(cosemObject.getAttribute(DEFAULT_ATTRIBUTE_ID));
```

Comment:
In most of the cases attribute id 2 (the DEFAULT_ATTRIBUTE_ID) will be used. But in other cases, we need to use another attribute id. Do we need a more general function as well?

Reply:
Added operations to retrieve default attributes for optional and non-optional finds.